src/Corax/Queries/AllEntriesMatch.cs

```diff
@@ -69,7 +69,7 @@ public unsafe int Fill(Span<long> matches)
 
                 while (results < matches.Length)
                 {
-                    var read = Container.GetEntriesInto(_entriesContainerId, ref _offset, _currentPage, matches, results, out _itemsLeftOnCurrentPage);
+                    var read = Container.GetEntriesInto(_entriesContainerId, ref _offset, _currentPage, matches[results..], out _itemsLeftOnCurrentPage);
                     if (read == 0)
```

Comment:
Here we should use the `_itemsLeftOnCurrentPage` somehow too to avoid retrying on `.GetEntriesInto()` multiple times too.